Update RequireResource to handle duplicate short names and proper fully qualified names (also make life a little better for windows file system contributors)

jhandel opened the pull request:
# Description

Updated clivalidation.RequireResourceType so that it can take either fully qualified names or short names.. and if a short name is used and there are duplicate short names display the list of matching short names.

I also added a .gitattributes to the following folders so that their test files would maintain lf line ends instead of crlf line ends when cloned or pulled into a windows filesystem.

* pkg/corperp/renderers/containers/testdata
* pkg/recipes/terraform/config/testdata
* pkg/recipes/terraform/testdata/.terraform/modules


## Type of change

This pull request fixes a bug in Radius and has an approved issue 

Fixes: #7070

This pull request fixes a bug in Radius but its not an approved issue

Fixes:  #7121

jhandel committed:
adding .gitattributes to testdata folders so that the sample data uses the same linefeed as the test expects

Signed-off-by: Josh <josh@liveoak.ws>

jhandel committed:
update RequireResourceType to handle full type names and ask for clarification if duplicates are found

Signed-off-by: Josh <josh@liveoak.ws>

vinayada1 commented on pkg/cli/clivalidation.go:
```diff
@@ -270,13 +270,21 @@ func RequireResourceType(args []string) (string, error) {
-		if strings.EqualFold(supportedType, resourceTypeName) {
-			return resourceType, nil
+		//check to see if the resource type is the correct short or long name.
+		if strings.EqualFold(supportedType, resourceTypeName) || strings.EqualFold(resourceType, resourceTypeName) {
```
Could you add a unit test for this change?

jhandel replied:
@vinayada1,  it doesn't look like any of these functions in clivalidation are being independantly unit tested.. Do you want me to test this in the resources module (its used by list, show and delete) or should we start a clivalidation_test.go file and start it with this test?

vinayada1 replied:
You are right! It would be great if you could start with a clivalidation_test.go file. Thanks!

vinayada1 commented:
@jhandel  - Thanks for your contribution.

jhandel committed:
Testing for RequireResourceTypeAndName
Added new clivalidation_test.go with a test for RequireResrouceType
Also Updated delete_test, list_test, and show_test in cli/resources so that the arguments tests for those commands would include testing for ambigous types

Signed-off-by: Josh <josh@liveoak.ws>

jhandel committed:
Merge branch 'main' into fix_7070
